packages/flutter/test/widgets/autocomplete_test.dart

```diff
@@ -647,6 +647,52 @@ void main() {
     );
   });
 
+  testWidgets('support asynchronous options builder', (WidgetTester tester) async {
+    final GlobalKey fieldKey = GlobalKey();
+    final GlobalKey optionsKey = GlobalKey();
+    late FocusNode focusNode;
+    late TextEditingController textEditingController;
+    late Iterable<String> lastOptions;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: RawAutocomplete<String>(
+            optionsBuilder: (TextEditingValue textEditingValue) async {
+              return kOptions.where((String option) {
```

Comment:
Could you add another test that returns a Future.delayed here?  I want to test that actually getting the options after a delay will work as expected.

I think you could also test the "out of order" case that I mentioned in my other comment too, if we think that's a useful test to have.  Like type in the field twice, and the first time return a Future.delayed with a long delay and the second time return quickly/instantly.

Reply:
Should I add the "out of order" case even if it wouldn't work as expected? As in, when the delayed request returns, it overrides the second quick request.